Cython/Compiler/Optimize.py

```diff
@@ -339,20 +339,22 @@ def _transform_reversed_iteration(self, node, reversed_function):
             PyrexTypes.CFuncTypeArg("s", Builtin.bytes_type, None)
             ])
 
-    def _transform_indexable_iteration(self, node, slice_node, reversed=False):
+    def _transform_indexable_iteration(self, node, slice_node, is_mutable, reversed=False):
         """In principle can handle any iterable that Cython has a len() for and knows how to index"""
         unpack_temp_node = UtilNodes.LetRefNode(
             slice_node.as_none_safe_node("'NoneType' is not iterable"),
-            may_hold_none=False
+            may_hold_none=False, is_temp=True
```

Comment:
There are helper methods `.coerce_to_simple()` and `.coerce_to_temp()` for this need.

Reply:
It's possible I'm missing something but: `is_temp` is part of the constructor interface for `LetRefNode` so I think it's a valid thing to set. In this case I'm setting it as true because the `LetNode` ensures that it does end up as a temp. If I don't set it then it gets coerced to a temp (again) elsewhere, and generates a lot of unnecessary increfs and decrefs before each use..

I suspect the really correct thing to do would be to get `LetNode` to mark it as a temp correctly. However, when I tried that it broke other code in confusing ways...

Reply:
Ah, yes, you're right. It should be set to `True`.

Reply:
I actually wonder why it's not always true – or at least in most cases… Something to revisit in the places where it's being used currently.